src/core/task_mngr.c

```diff
@@ -0,0 +1,195 @@
+#include "task_mngr.h"
+
+
+//this functions are inline because they are run only from task manager at one place.
+void run_systemtasks_preexecute(void);
+void run_systemtasks_postexecute(void);
+void run_calculate_priority(void);
+void run_usertasks(void);
+task_t* find_next_task();
+
+task_t* task_user_active = NULL;
+task_t* task_user_list[] = {[0 ... (TASK_USER_LIST_SIZE-1)] = NULL};
+task_t* task_system_preexecute[] = {[0 ... (TASK_SYSTEM_PREEXECUTE_LIST_SIZE-1)] = NULL};
+task_t* task_system_postexecute[]= {[0 ... (TASK_SYSTEM_POSTEXECUTE_LIST_SIZE-1)] = NULL};
+task_t* task_user_backproc[] = {[0 ... (TASK_USER_BACKPROC_LIST_SIZE-1)] = NULL};
+
+int task_user_couter = 0;
+int task_system_preexecute_couter = 0;
+int task_system_postexecute_couter = 0;
+int task_user_backproc_couter = 0;
+
+void task_mngr_run(){
+	print_header();
+
+	if(task_user_couter == 0 &&
+			task_system_postexecute_couter == 0 &&
+			task_system_preexecute_couter == 0){
+		error("there are not task in list");
+		return;
+	}
+	//*************** Task manager main loop ***********************
+	info("starting task manager!");
+	while(1){
+
+		// 1. run pre-executed system tasks from list
+		if(task_system_preexecute_couter > 0){
+			run_systemtasks_preexecute();
+		}
+
+		// 2. run currently active task. If there is no active task chose one with biggest priority.
+		if(task_user_couter > 0){
+			run_usertasks();
+		}
+
+		//3. run post-executed system tasks
+		if(task_system_postexecute_couter > 0){
+			run_systemtasks_postexecute();
+		}
+
+		for(int i = 0; i < 100000000; i++);
+	}
+}
+
+
+void run_usertasks(void){
+	// active task is active run his run callback
```

Comment:
Shouldn't this comment be on line 62?

The first `if` checks if there is an active task, no matter what state it is in.
The `if` on line 63 checks if the active task is in state active.